deepchem/models/keras_model.py

```diff
@@ -4,19 +4,26 @@
 import logging
 import os
 try:
-  from collections.abc import Sequence
+  from collections.abc import Sequence as SequenceCollection
 except:
-  from collections import Sequence
+  from collections import Sequence as SequenceCollection
 
 logger = logging.getLogger(__name__)
 
-from deepchem.data import NumpyDataset
+from deepchem.data import Dataset, NumpyDataset
+from deepchem.metrics import Metric
 from deepchem.models.losses import Loss
 from deepchem.models.models import Model
-from deepchem.models.optimizers import Adam
-from deepchem.trans import undo_transforms
+from deepchem.models.optimizers import Adam, Optimizer, LearningRateSchedule
+from deepchem.trans import Transformer, undo_transforms
 from deepchem.utils.evaluate import GeneratorEvaluator
 
+from typing import Any, Callable, Dict, Iterable, List, Optional, Sequence, Tuple, TypeVar, Union
```

Comment:
Should we add a `deepchem.types` module to gather all DeepChem types in one place?

Reply:
I think a `deepchem.types` module and/or some dedicated documentation explaining what the DeepChem type conventions are would be really helpful. Type annotations are great, but it's really easy to become inconsistent across modules. The `mypy` [cheatsheet](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html) might be a good starting point.

Reply:
Good idea.  How about `deepchem.utils.typing`?  I've already come across a couple of compound types that are likely to appear in a lot of places.

Reply:
Works great! It might also be worth adding a `docs/typing.rst` docs file that lays out DeepChem type conventions for newcomers

Reply:
What about a more general `coding.rst` file for coding conventions?  In addition to type annotations, it could also discuss unit tests, doc strings, code formatting, etc.

Reply:
Sounds like a great idea!